omnibus.rb

```diff
@@ -44,9 +44,14 @@
 fetcher_retries 3
 fetcher_read_timeout 120
 
-# We limit this to 10 workers to eliminate transient timing issues in the
-# way Ruby (and other components) compiles on some more esoteric *nixes.
-workers 10
+# Work around for solaris zones, provide a less-bad number of workers
+if solaris? && Ohai['cpu']['total'] == 0
+  workers 10
+end
```

Comment:
Now I'm really confused because the red code here says `workers 10` so you should be getting 10 workers everywhere, so how are you falling into the ncpus code and getting 255 workers on the solaris zones with the silly number of virtual cores?